lib/src/parsers/example-parser.spec.ts

```diff
@@ -0,0 +1,99 @@
+import { extractJSDocExamples } from "./example-parser";
+import { createProjectFromExistingSourceFile } from "../spec-helpers/helper";
+import {
+  getJsDoc,
+  parseTypeReferencePropertySignaturesOrThrow
+} from "./parser-helpers";
+import { JSDoc, TypeAliasDeclaration } from "ts-morph";
+import { StringType, Type, TypeKind } from "../types";
+
+type JSDocNodeMapping = {
+  ParsedExamples: JSDoc[];
+  MismatchedExampleAndIntegerType: JSDoc[];
+  MismatchedExampleAndStringWithQuotesType: JSDoc[];
+  [typeName: string]: JSDoc[];
+};
+describe("example-parser", () => {
+  describe("extractJSDocExamples", () => {
+    const jsDocs: JSDocNodeMapping = {} as JSDocNodeMapping;
+    const sourceFile = createProjectFromExistingSourceFile(
+      `${__dirname}/__spec-examples__/examples.ts`
+    ).file;
+    const typeValues = sourceFile.getTypeAliases();
+
+    typeValues.forEach((typeAlias: TypeAliasDeclaration) => {
+      const properties = parseTypeReferencePropertySignaturesOrThrow(
```

Comment:
Continuing on from above comments, I think its clearer and more consistent with other tests if each test relies on this `properties` array. i.e. at the beginning of each `test` we do something along the lines of:

```
properties.find(p => p.getName() === "property-with-example").getJsDocs()
```
We can extract this into a function:
```
function getJsDocsFromProperty(properties: PropertySignature, propertyName: string): JsDoc[] {
 ...
}
```
I think this allows additional tests / examples easier to add and is simpler to reason with.